File: montecarlo/montecarlopi.py
import random

totalpoints = int(1e7)
circlepoints = 0
for i in range(1, totalpoints):
    x = random.uniform(0, 1)
    y = random.uniform(0, 1)
    if x**2+y**2 <= 1:
        circlepoints += 1

pival = 4*circlepoints/totalpoints

print(pival)

File: montecarlo/montecarlopi.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int totalPoints = 10000000;
    int circlePoints = 0;
    
    srand(time(NULL));

    for (int i = 0; i < totalPoints; i++) {
        double x = (double)rand() / RAND_MAX; 
        double y = (double)rand() / RAND_MAX; 

        if (x * x + y * y <= 1) {
            circlePoints++;
        }
    }

    double piVal = 4.0 * circlePoints / totalPoints;
    printf("Estimated value of Pi: %f\n", piVal);

    return 0;
}

